src/mockgyver.erl

```diff
@@ -1244,23 +1249,23 @@ get_code(Mod) ->
             end
     end.
 
-mock_mod(UserAddedFAs, #modinfo{key=?modinfo_key(Mod),
-                                exported_fas=ExportedFAs,
-                                code=Bin,
-                                filename=Filename}) when is_binary(Bin) ->
-    OrigMod = reload_mod_under_different_name(Mod, Bin, Filename),
-    OrigHash = get_module_checksum(OrigMod),
+mock_mod(UserAddedFAs,
+         #modinfo{key=?modinfo_key(Mod), exported_fas=ExportedFAs}=Modinfo,
+         CacheModDelta) ->
+    ShadowMod = mod_name_as_shadow(Mod), % mod -> mod^
```

Comment:
Hmm.. I understand your point. I've changed to 'mocked' but I'm not really happy with it. Previously the code talked about 'orig' for the `mod^` but this is certainly not the original. Keeping track of both an OrigChecksum for the `mod^` module and another checksum for the on-disk 'original' module made me mix things up more than once. Similarly, I think 'mocked' means two things just the same way 'orig' did, and has the same potential for causing mix-ups.

> If the term shadow is introduced, I'd like it to be consistent.

Maybe this is the best way to go, to introduce an intermediary commit that renames things. If going this way, should it go in the beginning of the commit chain, or just before the last commit (that is the one that introduced the term 'shadow')

Reply:
Revisiting/unifying the naming would be great, probably good since this PR introduces more to keep track of. Not sure a checksum of both the "real" original and its ^ version is needed though?  Only a change in the former would invalidate the cache and trigger a remake of the ^.

As for where in the commit history, either is fine.

Reply:
>> Keeping track of both an OrigChecksum for the mod^ module
>
> Not sure a checksum of both the "real" original and its ^ version is needed though?

I know, but the code on master seems to keep a checksum of the ^ for the mocking mod cache if I read the code right, around [here](https://github.com/klajo/mockgyver/blob/31fb60d/src/mockgyver.erl#L1111) (the `OrigMod` is the ^) so I just kept doing the same, for better or worse. But then I needed to also really keep track of the on-disk original as well.
Anyway, maybe the checksum wasn't the only or even best example. I also need to keep track of both code and module name for both the original and the ^.

Reply:
> As for where in the commit history, either is fine.

I'll look into inserting some naming cleanup commit somewhere suitable.

Reply:
> > > Keeping track of both an OrigChecksum for the mod^ module
> > 
> > 
> > Not sure a checksum of both the "real" original and its ^ version is needed though?
> 
> I know, but the code on master seems to keep a checksum of the ^ for the mocking mod cache if I read the code right, around [here](https://github.com/klajo/mockgyver/blob/31fb60d/src/mockgyver.erl#L1111) (the `OrigMod` is the ^) so I just kept doing the same, for better or worse. But then I needed to also really keep track of the on-disk original as well.

Yeah, using the checksum of the ^ module to spot changes to the "original" module. Would have been nicer to do it on the proper orig module (Mod in that code). No idea why I didn't do that...

> Anyway, maybe the checksum wasn't the only or even best example. I also need to keep track of both code and module name for both the original and the ^.

No worries, I understood what you were getting at.

Code for the mocking module, the ^ module and the real module, and checksum for the real module I suppose.

Reply:
> > As for where in the commit history, either is fine.
> 
> I'll look into inserting some naming cleanup commit somewhere suitable.

Thank you so much!

Reply:
> Code for the mocking module, the ^ module and the real module, and checksum for the real module I suppose.

No probs, What are good names? Maybe 'mocking', 'shadow' and 'orig'? The ^ is unfortunately not a valid character in a variable name :) (unless maybe the [pinning eep](https://github.com/erlang/eep/blob/master/eeps/eep-0055.md) would get implemented, but then we'd be in trouble) I'm a little hesitant about 'real,' (not sure if it was meant as a term) the other ones are not really 'unreal' :)

(oops, accidentally marked this as resolved, so unresolving it again)

Reply:
^ = hat module ;-)

Mocking/mock, renamed and original? Or real instead of original. Yeah, everything is "real" in some sense, but...

Looked at some synonyms for real + original, the best were: true, actual, initial, authentic.  Original/real/actual, any of those will work for me.

As for the ^: Victim? Pushover?

Reply:
> Victim? Pushover?

:) Naming of things... I settled for mocking, renamed and original. (Halfway through, I came to think one could argue that it is the original module that is renamed. :) But I think it is reasonably clear anyway, so I went ahead.)

All is lined up now, I hope.